### PR TITLE
feat: Compact v0.30.0 grammar, highlights overhaul, and metadata fixes

### PR DESCRIPTION
**Pull request by adamreynolds-io**
## Summary

Major overhaul of the tree-sitter grammar, syntax highlighting, test coverage, documentation, and package metadata. Updates the grammar from Compact ~v0.17.0 to v0.30.0, fixes long-standing bugs, and brings the project to a reviewable baseline.

While this is potentially a better baseline, the tree-sitter grammar, at least, should be generated directly from the grammar declaration in the Compact compiler's `parser.ss`.

### Grammar changes

- Block comments (`/* ... */`)
- `$`-prefixed identifiers (matches Compact lexer spec)
- Single-quoted strings (enables `Opaque<'string'>`)
- String escape sequences (`\"`, `\'`, `\\`, `\n`, `\t`, `\r`, `\0`, `\uXXXX`, `\xXX`)
- Hex (`0xFF`), octal (`0o77`), binary (`0b1010`) number literals
- Assignments as expressions (right-associative `assign_expr`)
- Index access accepts expressions (not just `nat`)
- For-loop range bounds accept `tsize` (identifiers for generic params)
- Multi-binding `const` with `cbinding` rule (`const x = 1, y: Field = 2;`)
- Selective imports (`import { foo, bar as baz } from Module;`)
- Type aliases (`type U32 = Uint<32>;`, `new type Amount = Uint<64>;`)
- `nominal` field on `type_alias_declaration` for `new` keyword (mirrors compiler's `nominal?` flag)
- Bytes literals (`Bytes[0xFF, 0x00]`) with spread (`Bytes[...a, ...b]`)
- Array literal as named node (`array_literal`) with spread support
- `slice_term` built-in expression
- Assert as expression: `assert(condition, "message")`
- External contract body supports both `;` and `,` separators

### Bug fixes

- String literal regex now handles escape sequences (`\"`, `\n`, etc.)
- Struct body semicolons changed from terminator to delimiter style (trailing semicolon now optional)
- `wdecl` argument wrapping aligned with `cdefn`/`ecdecl_circuit`/`lconstructor`
- Typo fix: `struct_named_filed_initializer` → `struct_named_field_initializer`
- Repository URLs corrected in Cargo.toml, pyproject.toml, and tree-sitter.json (was `midnight-ntwrk/tree-sitter-compact`, now `midnightntwrk/compact-tree-sitter`)

### Highlights overhaul

- `<`/`>` operator captures now use named nodes `(greater_than)`/`(less_than)` to avoid conflict with `@punctuation.bracket`
- Member access highlighted as `@property` instead of `@function.call`
- Removed whole-node captures `(assert_term) @keyword` and `(bytes_literal) @constructor` that incorrectly colored child expressions
- Added highlights for new keywords: `type`, `new`, `from`, `slice`
- Added highlight for `type_name`

### Breaking AST changes

Downstream consumers matching these node names will need updates:

- **`assign_stmt` removed** — assignments now appear as `assign_expr` inside `expression_sequence_stmt`
- **`assert_stmt` removed** — assert is now `assert_term` (an expression term, not a statement)
- **`edecl` removed** — external circuit declarations deprecated in Compact 0.28
- **`struct_named_filed_initializer` renamed** to `struct_named_field_initializer` (typo fix)
- **`type_alias_declaration` has new `nominal` field** — the `new` keyword is now accessible via `field("nominal")`
- **`wdecl` argument field changed** — extra `field("args",...)` wrapper removed for consistency with other declaration rules

### Documentation

- README: added bindings table, installation, development instructions, project structure
- CHANGELOG: added full `[Unreleased]` section documenting all changes

### Test coverage

- 159 tests, 0 failures
- 6 new test corpus files: comments, functions, literals, modules, type-alias, bytes-and-spread
- Removed `external.txt` (edecl deprecated)
- Tests cover: selective imports, bytes literals, spread elements, slice expressions, multi-binding const, comma-separated external contract bodies, string escape sequences, empty arrays/bytes, trailing commas, for-loop expression limits
- Rust binding tests pass

### Validation

Parsed real Compact v0.30.0 source files from the compiler test suite with no regressions. Errors only in intentionally invalid files (reserved keyword tests, old assert syntax).

Co-authored-by: faculerena <7917088+faculerena@users.noreply.github.com>

### CODE REVIEW

**Comment by adamreynolds-io** [comment]
## Code Review Summary

**16 findings** across grammar.js, highlights.scm, and test corpus.

| Severity | Count | Fixed | Dismissed |
|----------|-------|-------|-----------|
| P2 | 6 | 5 | 1 |
| P3 | 7 | 6 | 1 |
| P4 | 3 | 0 | 3 |

### Fixed (P2)

1. **`_struct_body_semicolon` used terminator style** — changed to delimiter style so `struct S { a: T; b: U }` parses without requiring a trailing semicolon. Also fixed `_struct_body_comma` to use `commaSep1` consistently.
2. **`wdecl` had extra `field("args",...)` wrapper** — removed to match `cdefn`, `ecdecl_circuit`, and `lconstructor` argument patterns.
3. **highlights.scm: `member_access_expr (id)` captured as `@function.call`** — changed to `@property` (both property access and method calls were incorrectly highlighted as function calls).
4. **highlights.scm: `<`/`>` always resolved to `@punctuation.bracket`** — operator captures now use named nodes `(greater_than)`/`(less_than)` to avoid conflict with bracket captures.
5. **Test name "assert statement without message" was wrong** — renamed to "assert statement with comparison" (grammar always requires a message).

### Fixed (P3)

6. **`struct_named_filed_initializer` typo** — renamed to `struct_named_field_initializer` across grammar, highlights, and tests.
7. **`(assert_term) @keyword` redundant** — removed (colored entire node including condition and message).
8. **`(bytes_literal) @constructor` redundant** — removed (colored entire node including element expressions).
9. **`commaSep` JSDoc** — fixed "one or more" → "zero or more".
10. **Missing edge-case tests** — added: empty `[]`, empty `Bytes[]`, trailing comma in array/bytes literals.
11. **`for_stmt` limit variant untested** — added test for `for (const i of expr)` form.

### Dismissed

- **P2: `expr_seq` can't represent 2-element sequences** — verified against Compact compiler (`compiler/parser.ss:723-729`): this is intentional. `expr-seq-many` requires 3+ elements (tuple syntax). 2-element comma sequences are not valid Compact.
- **P3: Conflicts array has redundant subsets** — tested removing `[$.fun, $.term]` and `[$.fun, $.pattern]`; both are required by `tree-sitter generate` (tree-sitter treats each conflict declaration independently, not as set unions).
- **P4: Field labels inconsistent across test files** — informational, older tests use a different style. Not worth reformatting all tests.
- **P4: `assign_expr` accepts non-lvalue targets** — standard tree-sitter practice; semantic validation is the compiler's responsibility.
- **P4: Nominal vs structural type aliases produce identical corpus output** — anonymous tokens are inherently invisible in corpus format. The `nominal` field is verified in `node-types.json`.

### Verification

- `tree-sitter test`: **159/159 passing**
- `cargo test`: **2/2 passing**
- `tree-sitter parse` on real Compact v0.30.0 files: **0 errors**

File: queries/highlights.scm [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ; This file is part of compact-tree-sitter.
-; Copyright (C) 2025 Midnight Foundation
+; Copyright (C) 2026 Midnight Foundation
```
Comment (gilescope) [inline queries/highlights.scm]:
```suggestion
; Copyright (C) Midnight Foundation
```

File: Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ license = "Apache"
 readme = "README.md"
 keywords = ["incremental", "parsing", "tree-sitter", "compact"]
 categories = ["parser-implementations", "parsing", "text-editors"]
-repository = "https://github.com/midnight-ntwrk/tree-sitter-compact"
+repository = "https://github.com/midnightntwrk/compact-tree-sitter"
 edition = "2021"
```
Comment (gilescope) [inline Cargo.toml:11]: Time for?
```suggestion
edition = "2024"
```
